membrane_curvature/base.py

```diff
@@ -0,0 +1,126 @@
+from membrane_curvature.curvature import mean_curvature, gaussian_curvature
+
+
+from MDAnalysis.analysis.base import AnalysisBase
+import MDAnalysis as mda
+
+
+class MembraneCurvature(AnalysisBase):
+
+    r"""
+    MembraneCurvature is a tool to calculate membrane curvature.
+
+    Parameters
+    ----------
+    universe : Universe or AtomGroup
+        An MDAnalysis Universe object.
+    select : str or iterable of str, optional. Default: 'all'
+        The selection string of an atom selection to use as a
+        reference to derive a surface.
+    pbc : bool, optional. Default: 'True'
+        Apply periodic boundary conditions.
+    n_x_bins : int, optional. Deafult: '100'
+        Number of bins in grid in the `x` dimension.
+    n_y_bins : int, optional. Default: '100'
+        Number of bins in grid in the `y` dimension.
+    x_range : tuple of (float, float), optional. Deafult: (0, `universe.dimensions[0]`)
+        Range of coordinates (min, max) in the `x` dimension.
+    y_range : tuple of (float, float), optional. Deafult: (0, `universe.dimensions[1]`)
+        Range of coordinates (min, max) in the `y` dimension.
+
+    Attributes
+    ----------
+    results.z_surface : ndarray
+        Surface derived from atom selection in every frame.
+        Array of shape (`n_frames`, `n_x_bins`, `n_y_bins`)
+    results.mean_curvature : ndarray
+        Mean curvature associated to the surface.
+        Array of shape (`n_frames`, `n_x_bins`, `n_y_bins`)
+    results.gaussian_curvature : ndarray
+        Gaussian curvature associated to the surface.
+        Arrays of shape (`n_frames`, `n_x_bins`, `n_y_bins`)
+
+    Returns
+    -------
+    average_z_surface : ndarray
+        Average of the array elements in `z_surface`
+    average_mean_curvature :
+        Average of the array elements in `mean_curvature`
+    average_gaussian_curvature:
+        Average of the array elements in `gaussian_curvature`
+
+    """
+
+    def __init__(self, universe, select='all',
+                 n_x_bins=100, n_y_bins=100,
+                 x_range=None,
+                 y_range=None,
+                 pbc=True, **kwargs):
+
+        super().__init__(universe.universe.trajectory, **kwargs)
+        self.ag = universe.select_atoms(select)
+        self.pbc = pbc
+        self.n_x_bins = n_x_bins
+        self.n_y_bins = n_y_bins
+        self.x_range = x_range if x_range else (0, universe.dimensions[0])
+        self.y_range = y_range if y_range else (0, universe.dimensions[1])
+
+        # Raise if selection doesn't exist
+        if len(self.ag) == 0:
+            raise ValueError("Invalid selection. Empty AtomGroup.")
+
+        # Warning message if range doesn't cover entire dimensions of simulation box
+        for dim_string, dim_range, num in [('x', self.x_range, 0), ('y', self.y_range, 1)]:
+            if (dim_range[1] < universe.dimensions[num]):
+                warnings.warn(f"Grid range in {dim_string} do not cover entire "
```

Comment:
It's good practice to also log a warning using the [logging facility](https://docs.python.org/3/library/logging.html). See density analysis https://github.com/MDAnalysis/mdanalysis/blob/8f9f402fe65d6a90e88b60b6da735c130425f00e/package/MDAnalysis/analysis/density.py#L181-L183 and eg https://github.com/MDAnalysis/mdanalysis/blob/8f9f402fe65d6a90e88b60b6da735c130425f00e/package/MDAnalysis/analysis/density.py#L420-L424 as examples.